### ISSUE TITLE
Openapi will only include a single payload for identically named methods from different services

### ISSUE THREAD

**Issue by ccrossley** [goadesign/goa#1771]
## Expected Behavior
One of the following:
- An Error message pointing the user in the right direction
- Allow TypeName to be used within a method to manually resolve naming conflicts
- Openapi generated payload names would include the service in the name (ie: AddRequestBody -> ServiceOneAddRequestBody)

## Actual Behavior
The openapi doc will only have a single payload for methods of the same name, even if those methods are in different services. 

## Steps to Reproduce the Problem (including complete and simple design if relevant)

`goa gen` this design:
```
package design

import (
	. "goa.design/goa/http/design"
	. "goa.design/goa/http/dsl"
)

var _ = API("goa-bug-repros", func() {
	Title("GOA-Bug-Repros")
	Description("For Debugging")
	Server("http:/localhost:8080")
})

var _ = Service("ServiceOne", func() {

	HTTP(func() {
		Path("/1")
	})

	Method("add", func() {

		Payload(func() {

			TypeName("ServiceOneAddPlayload")

			Attribute("a")
			Attribute("b")
		})

		Result(String)

		HTTP(func() {
			POST("/")

			Response(StatusOK)
		})
	})
})

var _ = Service("ServiceTwo", func() {

	HTTP(func() {
		Path("/2")
	})

	Method("add", func() {

		Payload(func() {

			TypeName("ServiceTwoAddPlayload")

			Attribute("c", String)
			Attribute("d", String)
			Attribute("e", String)
		})

		Result(String)

		HTTP(func() {
			POST("/")

			Response(StatusOK)
		})
	})
})
```
Here is the openapi.json:

```
{
  "swagger": "2.0",
  "info": {
    "title": "GOA-Bug-Repros",
    "description": "For Debugging",
    "version": ""
  },
  "paths": {
    "/1": {
      "post": {
        "tags": [
          "ServiceOne"
        ],
        "summary": "add ServiceOne",
        "operationId": "ServiceOne#add",
        "parameters": [
          {
            "name": "ServiceOneAddPlayload",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/AddRequestBody"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "OK response.",
            "schema": {
              "type": "string"
            }
          }
        },
        "schemes": [
          "http"
        ]
      }
    },
    "/2": {
      "post": {
        "tags": [
          "ServiceTwo"
        ],
        "summary": "add ServiceTwo",
        "operationId": "ServiceTwo#add",
        "parameters": [
          {
            "name": "ServiceTwoAddPlayload",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/AddRequestBody"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "OK response.",
            "schema": {
              "type": "string"
            }
          }
        },
        "schemes": [
          "http"
        ]
      }
    }
  },
  "definitions": {
    "AddRequestBody": {
      "title": "AddRequestBody",
      "type": "object",
      "properties": {
        "a": {
          "type": "string",
          "example": "Voluptas soluta."
        },
        "b": {
          "type": "string",
          "example": "Eligendi et quas ea architecto."
        }
      },
      "example": {
        "a": "Quibusdam fuga rem.",
        "b": "Iste atque magnam itaque."
      }
    }
  }
}
```

Notice that only one payload ended up in the openapi spec and instead of using TypeName, it's called AddRequestBody.  Changing the method name to be unique across all Services does solve the problem, but requires verbosely naming endpoints with common functions, like 'add'.

...or it's me, if so, please help!

## Specifications

  - goa Version: v2.0.0-wip
  - Platform (Linux/Windows etc.): OSX 10.13.2

**Comment by nitinmohan87** [comment]
Nice catch! This definitely looks like a bug to me.

**Comment by ccrossley** [comment]
Would changing the payload names be an acceptable way to avoid name collisions?  Is there a more eloquent way to scope the payloads in json schema?

**Comment by raphael** [comment]
This is fixed by #1823 - thank you!
